src/flask/helpers.py

```diff
@@ -1151,3 +1155,30 @@ def is_ip(value):
             return True
 
     return False
+
+
+def run_async(func):
+    """Ensure that the async function is run on a different thread."""
+
+    @wraps(func)
+    def _wrapper(*args, **kwargs):
+        call_result = Future()
+        def _run():
+            loop = asyncio.new_event_loop()
+            try:
+                result = loop.run_until_complete(func(*args, **kwargs))
+            except Exception as error:
+                call_result.set_exception(error)
+            else:
+                call_result.set_result(result)
+            finally:
+                loop.close()
+
+        loop_executor = ThreadPoolExecutor(max_workers=1)
```

Comment:
@miguelgrinberg is this acting as a global? I had thought this would result in a new executor per call, which in the case of a threaded or gthreaded server would mean multiple executors.

Reply:
Oh, right! Sorry, I understand now. Yes, you are making a brand new executor each time. So yes, concurrency works, but the expense of running an async view function is fairly high, the executor, the thread and the loop need to be created each time.